lib/Solver/Solver.cpp

```diff
@@ -0,0 +1,363 @@
+    Config->setConfigurationOption(Option, llvm::StringRef(Value.to_string()));
+  }
+  return Config;
+}
+
+// Class SolverConstraintVisitor
+bool SolverConstraintVisitor::addConstraint(vara::feature::Constraint *C) {
+  if (C->accept(*this)) {
+    S->Solver->add(Z3ConstraintExpression);
+    return true;
+  }
+  return false;
+}
+
+bool SolverConstraintVisitor::visit(vara::feature::BinaryConstraint *C) {
+
+  bool LHS = C->getLeftOperand()->accept(*this);
+  z3::expr Left = Z3ConstraintExpression;
+  bool RHS = C->getRightOperand()->accept(*this);
+  if (!LHS || !RHS) {
+    return false;
+  }
+  switch (C->getKind()) {
+  case feature::Constraint::ConstraintKind::CK_ADDITION:
+    Z3ConstraintExpression = Left + Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_AND:
+    Z3ConstraintExpression = Left && Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_DIVISION:
+    Z3ConstraintExpression = Left / Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_EQUAL:
+    Z3ConstraintExpression = Left == Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_EQUIVALENCE:
+    Z3ConstraintExpression = z3::implies(Left, Z3ConstraintExpression) &&
+                             z3::implies(Z3ConstraintExpression, Left);
+    break;
+  case feature::Constraint::ConstraintKind::CK_EXCLUDES:
+    Z3ConstraintExpression = z3::implies(Left, !Z3ConstraintExpression);
+    break;
+  case feature::Constraint::ConstraintKind::CK_GREATER:
+    Z3ConstraintExpression = Left > Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_GREATER_EQUAL:
+    Z3ConstraintExpression = Left >= Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_IMPLIES:
+    Z3ConstraintExpression = z3::implies(Left, Z3ConstraintExpression);
+    break;
+  case feature::Constraint::ConstraintKind::CK_LESS:
+    Z3ConstraintExpression = Left < Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_LESS_EQUAL:
+    Z3ConstraintExpression = Left <= Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_MULTIPLICATION:
+    Z3ConstraintExpression = Left * Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_NOT_EQUAL:
+    Z3ConstraintExpression = Left != Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_OR:
+    Z3ConstraintExpression = Left || Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_SUBTRACTION:
+    Z3ConstraintExpression = Left - Z3ConstraintExpression;
+    break;
+  case feature::Constraint::ConstraintKind::CK_XOR:
+    Z3ConstraintExpression = z3::implies(Left, !Z3ConstraintExpression) &&
+                             z3::implies(Z3ConstraintExpression, !Left);
+    break;
+  default:
+    return false;
```

Comment:
```suggestion
    return false;
```
Should this case ever happen? Maybe we should make this clear and not handle the case silently?

Reply:
This case should only happen if new operators in constraints are supported.
Should we throw an exception of some kind here?

Reply:
We have two options here: a) remove the default case, normally we should get a warning if there is no default an a case is not handled (if not we could enable that warning) b) we go the less easily detectable route and just add an assert before the return, so that at least in debug builds we would find the case